server/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -82,7 +83,7 @@
      */
     static final int MAX_SOURCE_LENGTH_IN_TOSTRING = 2048;
 
-    private String type = MapperService.SINGLE_MAPPING_NAME;
+    private String type;
```

Comment:
It might be useful to add a comment to these files `IndexRequest`, `UpdateRequest`, and `DeleteRequest`) explaining the different states of the type parameter, as it is not very obvious to a newcomer. I really should've done this in my initial PRs that began to modify these files, apologies for punting on that.